SimTracker/TrackTriggerAssociation/interface/TTClusterAssociationMap.h

```diff
@@ -38,48 +49,38 @@ class TTClusterAssociationMap {
 
-  void setTTClusterToTrackingParticlesMap(std::map<edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> >,
-                                                   std::vector<edm::Ptr<TrackingParticle> > > aMap) {
-    clusterToTrackingParticleVectorMap = aMap;
-  }
-  void setTrackingParticleToTTClustersMap(
-      std::map<edm::Ptr<TrackingParticle>, std::vector<edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > > >
-          aMap) {
-    trackingParticleToClusterVectorMap = aMap;
-  }
+  void setTTClusterToTrackingParticlesMap(const MapClusToVecTP<T>& aMap) { clusterToTrackingParticleVectorMap = aMap; }
+  void setTrackingParticleToTTClustersMap(const MapTPToVecClus<T>& aMap) { trackingParticleToClusterVectorMap = aMap; }
 
-  /// Operations
-  std::vector<edm::Ptr<TrackingParticle> > findTrackingParticlePtrs(
-      edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aCluster) const;
-  edm::Ptr<TrackingParticle> findTrackingParticlePtr(
-      edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aCluster) const;
-  std::vector<edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > > findTTClusterRefs(
-      edm::Ptr<TrackingParticle> aTrackingParticle) const;
+  /// Get all TPs associated to a cluster
+  std::vector<TrackingParticlePtr> findTrackingParticlePtrs(TTClusterRefT<T> aCluster) const;
+  /// Get main TP associated to a cluster. (Non-NULL if isGenuine() below is true).
+  TrackingParticlePtr findTrackingParticlePtr(TTClusterRefT<T> aCluster) const;
 
-  /// MC Truth methods
-  bool isGenuine(edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aCluster) const;
-  bool isCombinatoric(edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aCluster) const;
-  bool isUnknown(edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > aCluster) const;
+  // Get all clusters associated to TP.
+  std::vector<TTClusterRefT<T>> findTTClusterRefs(TrackingParticlePtr aTrackingParticle) const;
+
+  ///--- Get quality of L1 cluster based on truth info.
+  /// (exactly 1 of following 3 functions is always true)
+
+  /// Cluster "genuine": i.e. cluster associated to exactly 1 TP.
+  /// (If other TPs are associated, but have in total < 1% of Pt of main TP,
+  ///  or if they are null, then they are neglected here).
+  bool isGenuine(TTClusterRefT<T> aCluster) const;
+  /// Cluster "unknown": i.e. not associated with any TP.
+  bool isUnknown(TTClusterRefT<T> aCluster) const;
+  /// Cluster is not "genuine" or "unknown".
+  bool isCombinatoric(TTClusterRefT<T> aCluster) const;
 
 private:
   /// Data members
-  std::map<edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> >, std::vector<edm::Ptr<TrackingParticle> > >
-      clusterToTrackingParticleVectorMap;
-  std::map<edm::Ptr<TrackingParticle>, std::vector<edm::Ref<edmNew::DetSetVector<TTCluster<T> >, TTCluster<T> > > >
-      trackingParticleToClusterVectorMap;
+  MapClusToVecTP<T> clusterToTrackingParticleVectorMap;
```

Comment:
member variable names should end in underscore

Reply:
done